packages/url/src/createEntryUrlBuilder.ts

```diff
@@ -1,26 +1,57 @@
+import { removeTrailingSlash } from '@288-toolkit/strings';
 import type { Maybe } from '@288-toolkit/types';
-import { uriToPath } from './uriToPath.js';
 
 export type Entry = {
-	language?: Maybe<string>;
-	uri?: Maybe<string>;
+	url?: Maybe<string>;
 };
 
 export type EntryUrlParams = {
-	localize: boolean;
+	/**
+	 * The site URL, i.e. the URL of your deployment.
+	 * The protocol and host are required and will be forced on the entry URL.
+	 */
 	siteUrl: string;
-	homeUri: string;
+	/**
+	 * Whether to remove the trailing slash from the pathname.
+	 * This is required because the home page in craft is an empty pathname, which,
+	 * when contacted with the site URL, will result in a trailing slash because Craft
+	 * forces the trailing slash on the site URL.
+	 *
+	 * @default true
+	 */
+	shouldRemoveTrailingSlash?: boolean;
 };
 
 /**
  * Creates a function that builds URLs for entries.
+ * Craft 5 compatible. (Not compatible with Craft 4)
  */
-export const createEntryUrlBuilder = ({ siteUrl, localize, homeUri }: EntryUrlParams) => {
+export const createEntryUrlBuilder = ({ siteUrl, shouldRemoveTrailingSlash = true }: EntryUrlParams) => {
+	if (!URL.canParse(siteUrl)) {
```

Comment:
`URL.canParse` support is at 89% (https://caniuse.com/?search=canParse), which is good but maybe a bit risky. Most notably, Safari 16 doesn't support it.

Reply:
Damn. I'll polyfill it then.

Reply:
See 476617dd2e and 76ad99a5ec